### PR TITLE
CleanUp primary_subdivisions and adif_modes before doing mig

### PR DESCRIPTION
**Pull request by int2001**
primary_subdivision-table has no PK.

this is no problemm but in combination with no downgrade here it could(!) produce dupes.
This one solves the problem, by:

- [x] Cleaning up table from (technical) dupes
- [x] Cleaning up table from logical dupes
- [x] Adding a UNIQUE_KEY
- [x] Take care of the UNIQUE_KEY

Same problem at adif_modes. This PR cleans up the table and adds a proper unique_key

### CODE REVIEW

File: application/migrations/231_adif_315.php [hunk 1]
```diff
@@ -8,542 +8,574 @@
...
+		// ############################################################################################################
+
+		/**
+		 * 2. Primary Subdivisions
+		 * 
+		 * https://adif.org/315/ADIF_315_annotated.htm#Primary_Administrative_Subdivision
+		 */
+
+		// Clean Up primary_subdivisions from dupes, since there's no PK!!
+		$this->db->query("CREATE TABLE tmp_psd AS SELECT DISTINCT ps.`adif`,ps.`state`,ps.`subdivision`,ps.`deprecated` FROM `primary_subdivisions` ps");
```
Comment (HB9HIL) [inline application/migrations/231_adif_315.php:141]: At this point in Code the column `ps.deprecated` does not exist. It will be created in Line 156

**Review by HB9HIL — APPROVED** [review]
tested sucessfully in docker

**Comment by AndreasK79** [comment]
Tested with no errors here. Set migration to 230, then upgraded. No errors in the log, and indexes were created.

**Review by AndreasK79 — APPROVED** [review]

**Review by phl0 — APPROVED** [review]
Works as designed here (not sure why I had errors before) ...

**Pull request merged**